### DISCUSSION TITLE
Any way to change the display order? (And other question).

### DISCUSSION THREAD

**Discussion by Samueru-sama** [fastfetch-cli/fastfetch#732] [Q&A] [answered]
Currently my fastfetch looks like this: 

![image](https://github.com/fastfetch-cli/fastfetch/assets/36420837/b21998f8-3e83-4c27-94dc-809f36524256)

On the display part, for some reason the Display order is DP-2, DP-1, DP-3 instead of DP-1, 2, 3. How can I fix that? 

I'm also fine if I could merge the 3 into a single line like I've done with OS and Kernel, though the text would need to shortened for them to fit. 

Also I changed the title of the Shell to display instead Interactive shell because I also want to display the name of the sh shell. Since I use the Dash shell for that I would like to print that info as well, any ideas on how I go do this as well?

**Comment by CarterLi** [comment, marked as answer]
> On the display part, for some reason the Display order is DP-2, DP-1, DP-3 instead of DP-1, 2, 3. How can I fix that?

Currently no way.

> I'm also fine if I could merge the 3 into a single line like I've done with OS and Kernel, though the text would need to shortened for them to fit.

`--display-compact-type original`

> Also I changed the title of the Shell to display instead Interactive shell because I also want to display the name of the sh shell. Since I use the Dash shell for that I would like to print that info as well, any ideas on how I go do this as well?

No support. Although it's possible if `/bin/sh` is a symlink.

**Reply by Samueru-sama** [reply]
Thank you, --display-compact-type original doesn't show the refresh rate but that is something I can add on my end.

/bin/sh is a symlink, pardon my ignorance but it is always a symlink isn't it?

**Reply by CarterLi** [reply]
> --display-compact-type original doesn't show the refresh rate

I'm considering adding it, but I'm not sure what's the preferred format.

> /bin/sh is a symlink, pardon my ignorance but it is always a symlink isn't it?

It isn't in macOS.

**Reply by Samueru-sama** [reply]
Oh, didn't know about that, my mind is too linux centric lol. 

About the format, For me if all the displays have the same refresh rate, it should only state `@XX Hz` once at the end, and if every display has a different refresh rate then the refresh rate is shown next the resolution like it normally does.

**Reply by Samueru-sama** [reply]
This is what I ended up doing, it is a terrible terrible hack and I feel ashamed posting it, but it works lol. I could not figure out how to get fastfetch to pipe two values at once so it is unnecessary being run twice, good thing it is fast! lol. 

```
    {
      "key": "Display",
      "type": "command",
      "text": "fastfetch -s display --display-compact-type original --pipe | awk '{ORS=\"\"}1; END{print\" @ \"}' | sed 's/Display //g' && fastfetch -s display --pipe | grep -oE '[0-9]+Hz' | awk 'NR==1{print $1}'"
    }
```

![image](https://github.com/fastfetch-cli/fastfetch/assets/36420837/75359a3e-023b-4f65-8108-6ea0f0a0673f)

**Reply by CarterLi** [reply]
Command module was not designed to be abused like this lol

Isnt submitting PR to get direct support a better idea?

**Reply by Samueru-sama** [reply]
You think I know enough to submit a PR? Thanks for that though.

Boy and this isn't the worse, I didn't like working with jq before and I had some very terrible scripts that used awk instead, I guess when all you have is a hammer everything looks like a nail lol. 

Edit: Just used the hammer again: 

```
    {
      "key": "Terminal",
      "type": "command",
      "text": "( fastfetch -s terminal --pipe | awk '{ORS=\"\"}1; END{print\" - \"}' && fastfetch -s terminalfont --pipe ) | sed 's/Terminal //g; s/Font/Font:/g'"
    },
```

![image](https://github.com/fastfetch-cli/fastfetch/assets/36420837/aa900778-81ea-4da5-beb0-deab0a3d5e1a)

**Comment by CarterLi** [comment]
`--display-compact-type original-with-refresh-rate` was added

**Reply by Samueru-sama** [reply]
Thank you, hopefully you didn't do this out of pity after seeing how I massacred the command module.
